src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -530,10 +530,18 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
 				this.appendData(args[0]);
 		}
 	}
+
 	private appendData(data: any[][]) {
+		const tableHasFocus = isAncestor(document.activeElement, <HTMLElement>this._inputContainer.nativeElement);
+		const wasFocused = tableHasFocus && this._table.grid.getDataLength() > 0 && this._table.grid.getActiveCell();
+
 		this._tableData.push(this.transformData(data, this.columns));
 		this.data = this._tableData.getItems().map(dataObject => Object.values(dataObject));
 		this.layoutTable();
+
+		if (wasFocused) {
+			this.focus();
```

Comment:
Instead of focusing the table in general I would think we'd want to keep the focus on the cell it was previously on